include/grpcpp/generic/generic_stub.h

```diff
@@ -123,30 +128,34 @@ class TemplatedGenericStub final {
     /// Setup and start a unary call to a named method \a method using
     /// \a context and specifying the \a request and \a response buffers.
     void UnaryCall(ClientContext* context, const std::string& method,
-                   const RequestType* request, ResponseType* response,
+                   const char* suffix_for_stats, const RequestType* request,
```

Comment:
Even though this is marked as non-API since it's in experimental, it is de-facto API (and is already API internally). As a result, would you consider not using this feature here? And, to look like generated stubs, why not pass in this feature as StubOptions in a new constructor?

Reply:
I totally agree that in isolation, this is a questionable design choice (and I second guessed it at least once during implementation).  IIUC GenericStub is analogous to AnonymousStub, which isn't fixed until call-time.  If the method is allowed to freely change (i.e. as a parameter here), the suffix should also.  Internally, this feature is needed by Extensible Stubs.  I *could* move this to an overloaded signature as a migration step though, if you'd feel more comfortable with that

Reply:
I understand, and that makes sense. That said, it does raise a general concern for me. This function is basically the mirror of the non-experimental UnaryCall function that's a few lines up and for which you pass in `suffix_for_stats=nullptr`. I expect that we'll be removing this function altogether in the near future. What's the plan for providing suffix_for_stats then?

Reply:
suffix_for_stats is meant to be optional (which is why I used const char* instead of std::string), so I think it does make perfect sense to provide 2 versions of these functions: one with a suffix and one without.  Alternatively, we could just provide this signature and have clients pass nullptr if they want to opt-out.  I was thinking to do that migration we'd need to start of with this as an experimental function

Reply:
Ok, I just want you to be aware that we will be removing this particular function after grpc/proposal#180 lands, but not right away so maybe there would be time to revive it as a different experimentalization. For now we can let it go as is.